nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/impl/ACPImpl.java

```diff
@@ -86,6 +88,10 @@ public void addACL(ACL acl) {
                     ACL local = getACL(ACL.LOCAL_ACL);
                     if (local != null) {
                         int i = acls.indexOf(local);
+                        ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+                        if (configurationService.isBooleanPropertyTrue(LEGACY_BEHAVIOR_PROPERTY)) {
```

Comment:
I'm quite concerned about calling the configuration service at such a low-level in the core. It's not completely cheap. Could you at least compute this in the constructor? But we'll have to find a way to hoist this higher, however I don't know how at the moment.

Reply:
You mean call the `ConfigurationService` in the ctor ?

Reply:
Yes.